src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -2132,14 +2082,31 @@ class ValuesClauseSegment(BaseSegment):
         SELECT * FROM VALUES (1,2) as t (a,b);
         SELECT * FROM (VALUES (1,2) as t (a,b));
         WITH a AS (VALUES 1,2) SELECT * FROM a;
-
     """
 
     type = "values_clause"
-
     match_grammar = Sequence(
         "VALUES",
-        Ref("DelimitedValues"),
+        Delimited(
+            OneOf(
+                Bracketed(
+                    Delimited(
+                        # DEFAULT keyword used in
+                        # INSERT INTO statement.
+                        "NULL",
+                        Ref("ExpressionSegment"),
```

Comment:
Again everything covered by expression segment.
Note this is essentially the same segment as we have for exasol as well so nothing special in spark3 that warrants a different parse structure.
IIRC the OG PR was open for a while and I did the general ANSI values clause fixes, making the OG PR redundant. but clearly it got merged without realising the issue was already fixed